### PR TITLE
V3

### PR DESCRIPTION
**Pull request by m-abdelwahab**
This PR does the following:
- return the branch data if it already exists
- returns two database URLs: pooled non-pooled

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -58,9 +61,52 @@ runs:
...
+        # Get the branch id by its name. We list all branches and filter by name
+        branch_id=$(curl --silent \
+            "https://console.neon.tech/api/v2/projects/${PROJECT_ID}/branches" \
+            --header "Accept: application/json" \
+            --header "Content-Type: application/json" \
+            --header "Authorization: Bearer ${API_KEY}" \
+            | jq -r .branches \
+            | jq -c '.[] | select(.name | contains("'${BRANCH_NAME}'")) .id' \
+            | jq -r \
+            ) \
```
Comment (raoufchebri) [inline action.yml:76]: I'm not sure why we are looking for the branch by name.

File: action.yml [hunk 2]
```diff
@@ -58,9 +61,52 @@ runs:
...
+        region=$(echo $endpoints | jq --raw-output '.endpoints[0].host' | cut -d'.' -f2)
+        cloud_provider=$(echo $endpoints | jq --raw-output '.endpoints[0].host' | cut -d'.' -f3)
+        host=$(echo $endpoints | jq --raw-output '.endpoints[0].host')
+        branch_id=$(echo $endpoints | jq --raw-output '.endpoints[0].branch_id')
+
         echo "host=${host}" >> $GITHUB_OUTPUT
+        echo "endpoint_id=${endpoint_id}" >> $GITHUB_OUTPUT
         echo "branch_id=${branch_id}" >> $GITHUB_OUTPUT
-        echo "db_url=postgres://${PGUSERNAME}:${PGPASSWORD}@${host}" >> $GITHUB_OUTPUT
+        echo "db_url=postgres://${PGUSERNAME}:${PGPASSWORD}@${endpoint_id}.${region}.${cloud_provider}.neon.tech" >> $GITHUB_OUTPUT
```
Comment (raoufchebri) [inline action.yml:96]: why not just use host?

File: action.yml (hunk 1 above)
Comment (m-abdelwahab) [reply action.yml:76]: the create branch action takes a name as a parameter, right?

So if the name already exists we can't create the branch

File: action.yml (hunk 2 above)
Comment (m-abdelwahab) [reply action.yml:96]: Just to be consistent with the other variable

**Comment by nicksrandall** [comment]
@m-abdelwahab I would love to use this. Any chance we can merge this?

**Pull request merged**